ui/src/app/machines/components/ActionFormWrapper/CloneForm/CloneFormFields/CloneNetworkTable/CloneNetworkTable.tsx

```diff
@@ -0,0 +1,251 @@
+      nic: NetworkInterface | null;
+    }) => {
+      if (link && !nic) {
+        [nic] = getLinkInterface(machine, link);
+      }
+      const fabric = getInterfaceFabric(machine, fabrics, vlans, nic, link);
+      const vlan = vlans.find(({ id }) => id === nic?.vlan_id);
+      const subnet = getInterfaceSubnet(
+        machine,
+        subnets,
+        fabrics,
+        vlans,
+        isAllNetworkingDisabled,
+        nic,
+        link
+      );
+      const nameDisplay = getInterfaceName(machine, nic, link);
+      const subnetDisplay = getSubnetDisplay(subnet, true);
+      const fabricDisplay = fabric?.name || "Unconfigured";
+      const vlanDisplay = getVLANDisplay(vlan);
+      const typeDisplay = getInterfaceTypeText(machine, nic, link, true);
+      const numaDisplay = (getInterfaceNumaNodes(machine, nic) || []).join(
+        ", "
+      );
+      const dhcpDisplay = getDHCPStatus(vlan, vlans, fabrics);
+
+      return {
+        columns: [
+          {
+            className: "name-col",
+            content: (
+              <DoubleRow
+                primary={nameDisplay}
+                primaryTitle={nameDisplay}
+                secondary={!isParent ? subnetDisplay : null}
+                secondaryTitle={subnetDisplay}
+              />
+            ),
+            "data-test": "name-subnet",
+          },
+          {
+            className: "fabric-col",
+            content: !isParent ? (
+              <DoubleRow
+                primary={fabricDisplay}
+                primaryTitle={fabricDisplay}
+                secondary={vlanDisplay}
+                secondaryTitle={vlanDisplay}
+              />
+            ) : null,
+            "data-test": "fabric-vlan",
+          },
+          {
+            className: "type-col",
+            content: (
+              <DoubleRow
+                primary={typeDisplay}
+                primaryTitle={typeDisplay}
+                secondary={numaDisplay}
+                secondaryTitle={numaDisplay}
+              />
+            ),
+            "data-test": "type-numa",
+          },
+          {
+            className: "dhcp-col",
+            content: !isParent ? dhcpDisplay : null,
+            "data-test": "dhcp",
+          },
+        ],
+      };
+    };
+    rows = [];
+    machine.interfaces.forEach((nic) => {
+      // Childless nics are always rendered normally. Next, if the nic has any
+      // parents they will be rendered right after, in order to show the
+      // parent-child hierarchy. Finally, any aliases are rendered after the
+      // parent-child grouping.
+      if (!isBondOrBridgeParent(machine, nic)) {
+        const firstLink = nic.links.length >= 1 ? nic.links[0] : null;
+        const row = generateRow({
+          isParent: false,
+          link: firstLink,
+          nic: firstLink ? null : nic,
+        });
+        rows.push(row);
+
+        nic.parents.forEach((parentId) => {
+          const parentNic = machine.interfaces.find(
```

Comment:
I think there's a `getBondOrBridgeParents` util which you could then loop through here.